src/db/scripts/fetchTags.js

```diff
@@ -0,0 +1,180 @@
+const isEqual = require('lodash/isEqual');
+const omit = require('lodash/omit');
+const uniq = require('lodash/uniq');
+const difference = require('lodash/difference');
+
+const GoogleSpreadsheet = require('google-spreadsheet');
+
+const doc = new GoogleSpreadsheet('1GxVDSZOWyGvtXa_dwmaRcjzWrXq_kjgXCmZEJVrgb10');
+
+const locales = ['be', 'en', 'ru'];
+
+const tagScopes = {
+  themes: { localized: ['title'], unlocalized: ['slug'] },
+  locations: { localized: ['title'], unlocalized: ['slug', 'image'] },
+  times: { localized: ['title'], unlocalized: ['slug'] },
+  personalities: {
+    localized: ['name', 'dates', 'description'],
+    unlocalized: ['slug', 'color', 'image'],
+  },
+};
+
+const parseSheet = (sheet, scopesLocalized = [], scopesUnlocalized = []) =>
+  new Promise(resolve => {
+    const result = [];
+    sheet.getRows({}, (error, rows) => {
+      rows.shift(); // This removes the "comment" row.
+      rows.forEach(row => {
+        const data = {};
+        scopesUnlocalized.forEach(scope => {
+          const value = row[scope];
+          if (value) {
+            set(data, [scope], row[scope]);
+          }
+        });
+        locales.forEach(locale => {
+          scopesLocalized.forEach(scope => {
+            const value = row[`${scope}.${locale}`];
+            if (value) {
+              set(data, [scope, locale], value);
+            }
+          });
+        });
+        result.push(data);
+      });
+      resolve(result);
+    });
+  });
+
+const getDB = async () => {
+  const topics = keyBy(await Topic.find({}).select('slug'), 'slug');
+  const tags = await Tag.find({});
+  return { topics, tags };
+};
+
+const getSheetsData = async info => {
+  const data = {};
+
+  const sheets = keyBy(info.worksheets, 'title');
+
+  await Promise.all(
+    Object.entries(tagScopes).map(([topic, details]) =>
+      parseSheet(sheets[topic], details.localized, details.unlocalized).then(tagRawData => {
+        data[topic] = tagRawData;
+      })
+    )
+  );
+
+  return data;
+};
+
+const validateSheetsData = sheetsData => {
+  const allTags = [];
+
+  Object.entries(sheetsData).forEach(([topic, tags]) => {
+    tags.forEach(tagData => {
+      if (!tagData.slug) {
+        throw new Error(`TOPIC: ${topic}, SLUG is missing for one of the objects`);
+      }
+      const { error } = Joi.validate(omit(tagData, ['slug']), TAG_CONTENT_SCHEMA[topic]);
+      if (error !== null) {
+        throw new Error(`TOPIC: ${topic}, SLUG: ${tagData.slug}, ${error}`);
+      }
+    });
+
+    const topicTags = tags.map(({ slug }) => slug);
+    allTags.push(...topicTags);
+
+    console.log(`  [OK] ${topic} : `, topicTags);
+  });
+
+  if (allTags.length !== uniq(allTags).length) {
+    throw new Error('it seems like there are slug duplicates in the Spreadsheet');
+  }
+
+  return allTags;
+};
+
+const updateTags = async (sheetsData, db) => {
+  const promises = [];
+
+  Object.entries(sheetsData).forEach(([topic, tags]) => {
```

Comment:
please use `map` instead `forEach` & `push`

Reply:
The problem was the arrays were nested.
Rewrote it with `map`-s and `flatten`.